# COVID-19 Related Queries for BTE Use Case Exploration 

In [42]:
import pandas as pd
## Load BTE
from biothings_explorer.user_query_dispatcher import FindConnection
from biothings_explorer.hint import Hint
ht = Hint()

## Functions that will be used
# Check for every intermediate node type in Predict funciton
def predict_many(input_object, intermediate_node_list, output_type):
    df_list = []
    for inter in intermediate_node_list:
        try: 
            print("Intermediate Node type running:")
            print(inter)
            fc = FindConnection(input_obj=input_object, output_obj=output_type, intermediate_nodes=[inter])
            fc.connect(verbose=False)
            df = fc.display_table_view()
            rows = df.shape[0]
            if(rows > 0):
                df_list.append(df)
        except:
            print("FAILED")
    if(len(df_list) > 0):
        return pd.concat(df_list)
    else:
        return None
    
# all intermediate node types

node_type_list = (['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', 
                   'MolecularActivity', 'BiologicalProcess', 'CellularComponent', 
                   'Pathway', 'AnatomicalEntity', 'PhenotypicFeature'])

## 1.1 TMPRSS2 -> NAFAMOSTAT
### In what ways can the TMPRSS2 Gene be connected to NAFAMOSTAT?
### 1.1 Query 

In [43]:
tmprss2 = ht.query("TMPRSS2")['Gene'][0]

all_tmprss2_predictions = predict_many(tmprss2,node_type_list,'ChemicalSubstance')

Intermediate Node type running:
Gene
Intermediate Node type running:
SequenceVariant
Intermediate Node type running:
ChemicalSubstance
Intermediate Node type running:
Disease
API 8.7 pharos failed
API 8.4 pharos failed
API 8.15 pharos failed
API 8.1 pharos failed
API 8.14 pharos failed
API 8.12 pharos failed
API 8.2 pharos failed
API 8.6 pharos failed
API 8.13 pharos failed
API 8.10 pharos failed
API 8.3 pharos failed
API 8.5 pharos failed
API 8.11 pharos failed
API 8.8 pharos failed
API 8.9 pharos failed
API 8.17 pharos failed
API 8.22 pharos failed
API 8.16 pharos failed
API 8.18 pharos failed
API 8.21 pharos failed
API 8.23 pharos failed
API 8.20 pharos failed
API 8.25 pharos failed
API 8.19 pharos failed
API 8.36 pharos failed
API 8.31 pharos failed
API 8.32 pharos failed
API 8.33 pharos failed
API 8.26 pharos failed
API 8.35 pharos failed
API 8.29 pharos failed
API 8.34 pharos failed
API 8.37 pharos failed
API 8.24 pharos failed
API 8.27 pharos failed
API 8.30 pharos failed
API 8.

### 1.1 Results table 

In [44]:
## FUT 175 (Futhan) is another name for Nafamostat 
all_tmprss2_predictions[(all_tmprss2_predictions["output_name"] == "FUT 175")]

input input_type                      pred1 pred1_source  \
30984  TMPRSS2       Gene  physically_interacts_with         None   
2405   TMPRSS2       Gene                 related_to      scibite   
2406   TMPRSS2       Gene                 related_to     scigraph   
2407   TMPRSS2       Gene  physically_interacts_with     drugbank   
43018  TMPRSS2       Gene                 related_to      scibite   
43019  TMPRSS2       Gene                 related_to          CTD   

                         pred1_api  \
30984                  BioLink API   
2405    Automat CORD19 Scibite API   
2406   Automat CORD19 Scigraph API   
2407               MyChem.info API   
43018   Automat CORD19 Scibite API   
43019                      CTD API   

                                       pred1_pubmed         node1_type  \
30984                                      32296183               Gene   
2405                                           None  ChemicalSubstance   
2406                                           None  ChemicalSubstance   
2407                                           None  ChemicalSubstance   
43018                                          None            Disease   
43019  24027332|24227843|27550352|27733646|28778717            Disease   

                           node1_name             node1_id  \
30984                             TNF        NCBIGene:7124   
2405                         CAMOSTAT        name:CAMOSTAT   
2406                         CAMOSTAT        name:CAMOSTAT   
2407                         CAMOSTAT        name:CAMOSTAT   
43018  CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   
43019  CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   

                           pred2 pred2_source                   pred2_api  \
30984  physically_interacts_with         None                   DGIdb API   
2405        negatively_regulates       SEMMED         SEMMED Chemical API   
2406        negatively_regulates       SEMMED         SEMMED Chemical API   
2407        negatively_regulates       SEMMED         SEMMED Chemical API   
43018                 related_to      scibite  Automat CORD19 Scibite API   
43019                 related_to      scibite  Automat CORD19 Scibite API   

      pred2_pubmed        output_type output_name     output_id  
30984         None  ChemicalSubstance     FUT 175  name:FUT 175  
2405       9322090  ChemicalSubstance     FUT 175  name:FUT 175  
2406       9322090  ChemicalSubstance     FUT 175  name:FUT 175  
2407       9322090  ChemicalSubstance     FUT 175  name:FUT 175  
43018         None  ChemicalSubstance     FUT 175  name:FUT 175  
43019         None  ChemicalSubstance     FUT 175  name:FUT 175

## 1.2 Question: Is the only way to get from TMPRSS2 to Blood Coagulation through Nafamostat (Fut 175)? 

In [48]:
tmprss2_to_blood_coagulation = predict_many(tmprss2,node_type_list,'BiologicalProcess')

Intermediate Node type running:
Gene
Intermediate Node type running:
SequenceVariant
Intermediate Node type running:
ChemicalSubstance
Intermediate Node type running:
Disease
Intermediate Node type running:
MolecularActivity
Intermediate Node type running:
BiologicalProcess
Intermediate Node type running:
CellularComponent
Intermediate Node type running:
Pathway
Intermediate Node type running:
AnatomicalEntity
Intermediate Node type running:
PhenotypicFeature


#### TMPRSS2 to Blood Coagulation through any node type:

In [50]:
tmprss2_to_blood_coagulation[tmprss2_to_blood_coagulation['output_name'].str.contains('COAGULATION', regex=False)]

input input_type                      pred1  \
4950   TMPRSS2       Gene  physically_interacts_with   
8807   TMPRSS2       Gene  physically_interacts_with   
8808   TMPRSS2       Gene  physically_interacts_with   
8809   TMPRSS2       Gene  physically_interacts_with   
8810   TMPRSS2       Gene  physically_interacts_with   
364    TMPRSS2       Gene                 related_to   
1117   TMPRSS2       Gene                 related_to   
1118   TMPRSS2       Gene                 related_to   
1119   TMPRSS2       Gene                 related_to   
1120   TMPRSS2       Gene                 related_to   
5379   TMPRSS2       Gene                     causes   
5380   TMPRSS2       Gene                 related_to   
5381   TMPRSS2       Gene                     causes   
5382   TMPRSS2       Gene                 related_to   
5383   TMPRSS2       Gene                 related_to   
5384   TMPRSS2       Gene                 related_to   
5385   TMPRSS2       Gene                     treats   
5386   TMPRSS2       Gene                 related_to   
5387   TMPRSS2       Gene                    affects   
5388   TMPRSS2       Gene                     treats   
5389   TMPRSS2       Gene                 related_to   
5390   TMPRSS2       Gene                    affects   
5391   TMPRSS2       Gene                 related_to   
5392   TMPRSS2       Gene                 related_to   
5393   TMPRSS2       Gene                 related_to   
5394   TMPRSS2       Gene                 related_to   
5395   TMPRSS2       Gene                 related_to   
5396   TMPRSS2       Gene                 related_to   
5397   TMPRSS2       Gene                 related_to   
5398   TMPRSS2       Gene                 related_to   
5399   TMPRSS2       Gene                 related_to   
5400   TMPRSS2       Gene                 related_to   
5401   TMPRSS2       Gene                 related_to   
11905  TMPRSS2       Gene                 related_to   
12746  TMPRSS2       Gene                 related_to   
12747  TMPRSS2       Gene                 related_to   
12748  TMPRSS2       Gene                 related_to   
12749  TMPRSS2       Gene                 related_to   
907    TMPRSS2       Gene     functional_association   
97     TMPRSS2       Gene     functional_association   
98     TMPRSS2       Gene     functional_association   
330    TMPRSS2       Gene                 related_to   

                          pred1_source                    pred1_api  \
4950                            SEMMED              SEMMED Gene API   
8807                            SEMMED              SEMMED Gene API   
8808                              None                  BioLink API   
8809                              None                  BioLink API   
8810                              None                  BioLink API   
364                           scigraph  Automat CORD19 Scigraph API   
1117                           scibite   Automat CORD19 Scibite API   
1118                          scigraph  Automat CORD19 Scigraph API   
1119                           scibite   Automat CORD19 Scibite API   
1120                          scigraph  Automat CORD19 Scigraph API   
5379                            SEMMED              SEMMED Gene API   
5380                            SEMMED              SEMMED Gene API   
5381                            SEMMED              SEMMED Gene API   
5382                            SEMMED              SEMMED Gene API   
5383                            SEMMED              SEMMED Gene API   
5384                            SEMMED              SEMMED Gene API   
5385                            SEMMED              SEMMED Gene API   
5386                            SEMMED              SEMMED Gene API   
5387                            SEMMED              SEMMED Gene API   
5388                            SEMMED              SEMMED Gene API   
5389                            SEMMED              SEMMED Gene API   
5390                            SEMMED

#### TMPRSS2 to Blood Coagulation through any ChemicalSubstance type node:

In [55]:
tmprss2_to_blood_coagulation[(tmprss2_to_blood_coagulation['node1_type'] == "ChemicalSubstance") & (tmprss2_to_blood_coagulation['output_name'].str.contains('COAGULATION', regex=False))]

input input_type       pred1 pred1_source  \
364   TMPRSS2       Gene  related_to     scigraph   
1117  TMPRSS2       Gene  related_to      scibite   
1118  TMPRSS2       Gene  related_to     scigraph   
1119  TMPRSS2       Gene  related_to      scibite   
1120  TMPRSS2       Gene  related_to     scigraph   

                        pred1_api pred1_pubmed         node1_type  \
364   Automat CORD19 Scigraph API         None  ChemicalSubstance   
1117   Automat CORD19 Scibite API         None  ChemicalSubstance   
1118  Automat CORD19 Scigraph API         None  ChemicalSubstance   
1119   Automat CORD19 Scibite API         None  ChemicalSubstance   
1120  Automat CORD19 Scigraph API         None  ChemicalSubstance   

       node1_name           node1_id       pred2  \
364   CHEBI:35222  CHEBI:CHEBI:35222  related_to   
1117      FUT 175       name:FUT 175     affects   
1118      FUT 175       name:FUT 175     affects   
1119      FUT 175       name:FUT 175    disrupts   
1120      FUT 175       name:FUT 175    disrupts   

                         pred2_source            pred2_api pred2_pubmed  \
364   Translator Text Mining Provider    CORD Chemical API         None   
1117                           SEMMED  SEMMED Chemical API      3443413   
1118                           SEMMED  SEMMED Chemical API      3443413   
1119                           SEMMED  SEMMED Chemical API      3443413   
1120                           SEMMED  SEMMED Chemical API      3443413   

            output_type        output_name               output_id  
364   BiologicalProcess        COAGULATION           GO:GO:0050817  
1117  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION  
1118  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION  
1119  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION  
1120  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION

1.2 NOTES: Tables show that TMPRSS2 -> coagulation can also be found through 'CHEBI:35222' Chemical Substance, as well as through many other node types that are not drugs

## 1.3 TMPRSS2 -> Camostat
### In what ways can the TMPRSS2 Gene be connected to Camostat?

In [56]:
all_tmprss2_predictions[all_tmprss2_predictions["output_name"] == "CAMOSTAT"]

input input_type                      pred1 pred1_source  \
3172   TMPRSS2       Gene       negatively_regulates       SEMMED   
3173   TMPRSS2       Gene  physically_interacts_with       SEMMED   
3174   TMPRSS2       Gene  physically_interacts_with         None   
3175   TMPRSS2       Gene  physically_interacts_with         None   
3176   TMPRSS2       Gene  physically_interacts_with         None   
42864  TMPRSS2       Gene                 related_to      scibite   
42865  TMPRSS2       Gene                 related_to     scigraph   
42866  TMPRSS2       Gene                 related_to          CTD   
42867  TMPRSS2       Gene                 related_to      DISEASE   
42868  TMPRSS2       Gene                 related_to      scibite   
42869  TMPRSS2       Gene                 related_to          CTD   
42870  TMPRSS2       Gene                 related_to      scibite   
42871  TMPRSS2       Gene                 related_to     scigraph   
42872  TMPRSS2       Gene                 related_to      scibite   
42873  TMPRSS2       Gene                 related_to      scibite   
42874  TMPRSS2       Gene                 related_to      scibite   
42875  TMPRSS2       Gene                 related_to     scigraph   

                         pred1_api  \
3172               SEMMED Gene API   
3173               SEMMED Gene API   
3174                   BioLink API   
3175                   BioLink API   
3176                   BioLink API   
42864   Automat CORD19 Scibite API   
42865  Automat CORD19 Scigraph API   
42866                      CTD API   
42867                 DISEASES API   
42868   Automat CORD19 Scibite API   
42869                      CTD API   
42870   Automat CORD19 Scibite API   
42871  Automat CORD19 Scigraph API   
42872   Automat CORD19 Scibite API   
42873   Automat CORD19 Scibite API   
42874   Automat CORD19 Scibite API   
42875  Automat CORD19 Scigraph API   

                                       pred1_pubmed node1_type  \
3172                                       25221645       Gene   
3173                                       25520876       Gene   
3174                                       21068237       Gene   
3175                                           None       Gene   
3176                                           None       Gene   
42864                                          None    Disease   
42865                                          None    Disease   
42866                    22496216|24227843|30626688    Disease   
42867                                          None    Disease   
42868                                          None    Disease   
42869  24027332|24227843|27550352|27733646|28778717    Disease   
42870                                          None    Disease   
42871                                          None    Disease   
42872                                          None    Disease   
42873                                          None    Disease   
42874                                          None    Disease   
42875                                          None    Disease   

                                       node1_name             node1_id  \
3172                                     C1710301        UMLS:C1710301   
3173                                     C1710301        UMLS:C1710301   
3174                                         ACE2       NCBIGene:59272   
3175                                      TMPRSS4       NCBIGene:56649   
3176                                      TMPRSS4       NCBIGene:56649   
42864           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42865           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42866           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42867           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42868              CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   
42869              CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   
4287